include/mppic/tools/utils.hpp

```diff
@@ -99,6 +99,29 @@ inline bool withinPositionGoalTolerance(
   return false;
 }
 
+inline bool withinPositionGoalTolerance(
+  const float & pose_tolerance,
+  const geometry_msgs::msg::Pose & robot,
+  const models::Path & path)
+{
+  const auto goal_idx = path.x.shape(0);
```

Comment:
```suggestion
  const auto goal_idx = path.x.shape(0) - 1;
```

Reply:
This is based on the `withinPositionGoalTolerance` implementation above that also does just `shape(0)`. Should I update that one too?

Reply:
Yep, that's a bug too.